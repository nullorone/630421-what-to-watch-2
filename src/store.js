import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import {reducer} from "./reducer";
import {compose} from "recompose";
import createApi from "./api";

const api = createApi((...args) => store.dispatch(...args));
/* eslint-disable no-underscore-dangle */
const store = createStore(
    reducer,
    compose(
        applyMiddleware(thunk.withExtraArgument(api)),
        window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
    ),
);
/* eslint-enable */

export default store;
