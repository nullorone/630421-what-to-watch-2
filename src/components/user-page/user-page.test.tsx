import * as React from "react";
import renderer from "react-test-renderer";
import UserPage from "./user-page";
import store from "../../reducer/store";
import {Provider} from "react-redux";

jest.mock(`../sign-in/sign-in`, () => `SignIn`);

describe(`Make snapshot`, () => {
  it(`Get snapshot component`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <UserPage/>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
