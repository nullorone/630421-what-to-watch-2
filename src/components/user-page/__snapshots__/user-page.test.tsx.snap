// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot Get snapshot component 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <Logo
      light={false}
    />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <SignIn
    onFormSubmitClick={[Function]}
  />
  <footer
    className="page-footer"
  >
    <Logo
      light={true}
    />
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
